### DISCUSSION TITLE
Relation Data displayed as a table

### DISCUSSION THREAD

**Discussion by rflstrmn** [Budibase/budibase#5593] [Help] [answered]
Hello!

I'm new to Budibase and I am trying to find a way to display linked records of a record in a table.

For a better idea, I have many teachers, each teacher is linked to multiple courses. The courses have multiple data like, date, room, attendees etc. Now when I select a particular teacher on a screen with the link /teacher/:id the name of the teacher appears there and a "relationship picker". Now I would like to output a table below that lists all the courses the teacher is linked to, including the course data, such as date, room, participants, etc.

Is there a way to do this? Thanks for any help!

Kind regards

**Comment by melohagan** [comment, marked as answer]
Hi @rflstrmn 

You can do this by using the relationship as a data source, similar to: https://docs.budibase.com/docs/fetching-data#display-user-specific-data

Unlike the Current User however, which is global, you will need to add a second data provider.

So the structure would be:

> Data Provider for Teachers (filtered on teacher id)
> > Repeater
> > > Data Provider (select your courses relationship here e.g. Repeater.Teacher.Courses)
> > > > Table (using the data provider above)

**Reply by rflstrmn** [reply]
Hi @melohagan 

many, many thanks! That was exactly what I was looking for. Intuitively I've almost got it right, I just linked the nested data provider (Course) directly to the course table.

Thanks that worked flawlessly!

**Reply by diedder** [reply]
hi @melohagan,
is it also possible to add a second table with negated output? My appoach is to have two dataables: qustions and questionkatalogs which collect's different questions of the whole question pool. To edit the questionkatalog two tables would be nice, one which shows all questions already in this selected katalog (relationship present) and the other ohne with all questions without existing relation.
br, diedder

**Reply by melohagan** [reply]
Hi @diedder 

What you could try, is to add a **static** (so it can be filtered) formula. Use the expression `{{ length Product }}` to get the number of relationships: 
<img width="471" alt="Screenshot 2022-05-29 at 20 35 43" src="https://user-images.githubusercontent.com/101575380/170888455-e114e13b-aed1-40fd-8617-9491c8433532.png">

And then you can simply have a filter where `NumberOfRelationships === 0` for the "negated" scenario.

**Reply by diedder** [reply]
Hi @melohagan 
I think I described not precisely... I don't want't to filter all data without relationship rather than all data whicht has not the relation as all the items in the upper list. Just like a filter Relationship != RelationXY but filtering on relations seem's not possible.  I'll be explaining my project vision in a separate thead more in detail. 
Thank you anyway!
